app/workers/historical_data_deletion_worker.rb

```diff
@@ -0,0 +1,81 @@
+class HistoricalDataDeletionWorker < ApplicationWorker
+  def perform
+    # cascades matched content changes
+    delete_and_log(ContentChange) { ContentChange.where("created_at < ?", retention_period) }
+
+    # cascades matched messages
+    delete_and_log(Message) { Message.where("created_at < ?", retention_period) }
+
+    # cascades digest run subscribers
+    delete_and_log(DigestRun) { DigestRun.where("created_at < ?", retention_period) }
+
+    # allows subscriber lists and subscribers to be deleted
+    delete_and_log(Subscription) { Subscription.where("ended_at < ?", retention_period) }
+
+    # cascades matched content changes and matched messages
+    delete_and_log(SubscriberList) { SubscriberList.where(id: empty_subscriber_list_ids + historic_subscriber_list_ids) }
+
+    # cascades digest run subscribers and emails
+    delete_and_log(Subscriber) { historic_subscribers }
+  end
+
+private
+
+  def retention_period
+    1.year.ago
+  end
+
+  def delete_and_log(model)
+    run_with_advisory_lock(model, "delete") do
```

Comment:
I don't think we need advisory locks for this, since these are single query they already run with transaction isolation.